canonical/prompting-client
# debug: add logging to the read script

This PR is based on #80, and is really just for debugging, probably should not be merged.

I'm convinced that the hanging on read tests is a result of the test runtime, rather than something in snapd. I'm not sure why the change to record notices in goroutines would have any effect on this, but the functionality of snapd and the test snap are correct either way.

To test, run the following in one terminal:
```
make prepare-vm
make integration-tests CASES=happy_path_read_single::allow
```

In another terminal (since the former will likely hang), run the following:
```
lxc exec aa-testing -- cat /home/ubuntu/snap/aa-prompting-test/common/log.txt
```

This will print out what is defined in the `read.sh` to be written to the log file, including the fact that `cat` successfully (and synchronously) read the contents of the file after being allowed by the prompt, and that `cat` is no longer running, and that the whole `read.sh` script finished running, regardless of what `tokio` claims :)
